socialhome/frontend/components/streams/Stream.vue

```diff
@@ -31,25 +32,30 @@
     </div>
 </template>
 
-
 <script>
 import Vue from "vue"
 
 import VueScrollTo from "vue-scrollto"
 
-import "frontend/components/StreamElement.vue"
-import PublicStampedElement from "frontend/components/stamped_elements/PublicStampedElement.vue"
-import FollowedStampedElement from "frontend/components/stamped_elements/FollowedStampedElement.vue"
-import TagStampedElement from "frontend/components/stamped_elements/TagStampedElement.vue"
-import ProfileStampedElement from "frontend/components/stamped_elements/ProfileStampedElement.vue"
-import "frontend/components/LoadingElement.vue"
-
 import {streamStoreOperations} from "frontend/stores/streamStore"
 
+import "frontend/components/streams/StreamElement.vue"
+import PublicStampedElement from "frontend/components/streams/stamped_elements/PublicStampedElement.vue"
+import FollowedStampedElement from "frontend/components/streams/stamped_elements/FollowedStampedElement.vue"
+import TagStampedElement from "frontend/components/streams/stamped_elements/TagStampedElement.vue"
+import ProfileStampedElement from "frontend/components/streams/stamped_elements/ProfileStampedElement.vue"
+import "frontend/components/streams/LoadingElement.vue"
+
 
 Vue.use(VueScrollTo)
 
 export default Vue.component("stream", {
+    // TODO: Seperate Stream.vue into TagStream.vue, GuidProfile.vue and UsernameProfile.vue, etc. in the future
```

Comment:
The `/u/username/` is not actually a real page, it just renders the real profile stream, if you look at what happens in the backend, so at least these should just use the same code. In general, I'm not sure we should start separating the code of different streams too much since other than the stamped element part they are identical in function.

Reply:
Sure, but you have one difference which is the `props` attribute. `guid`, or `username` or `tagName`. Most of the code will be identical but we can use Vue's extension system to create a class hierarchy. Eventually, the stamped element's code must be merged into theses new stream classes along with the new Stream classes. It's a bit difficult to explain without throwing some code.

Reply:
As long as we avoid code duplication I'm fine :+1: 

There will be many new types of streams so I would want the code to stay unified to avoid having to change each stream file when adding say a filter or other minor thing.